modules/enterprise/server/jar/src/main/java/org/rhq/enterprise/server/configuration/ConfigurationManagerBean.java

```diff
@@ -2652,6 +2653,34 @@ private void handlePDS(final Subject subject, Resource resource, Resource baseRe
             if (filter != null)
                 filterPattern = Pattern.compile(filter);
 
+            Predicate<Resource> baseResourcePredicate = null;
+            if (expressionScope == PropertyOptionsSource.ExpressionScope.BASE_RESOURCE) {
+                if (expression.contains(";")) {
```

Comment:
As expression.indexOf(";") is done so often in this codepart, wouldn't it make sense to do it once here?

```
int halfCommaIndex = expression.indexOf(";");
if(halfCommaIndex > -1) { 
.. processing code
}
```

Easier to read and change